src/OpenTelemetry.Exporter.InMemory/InMemoryExporterMetricsExtensions.cs

```diff
@@ -75,15 +93,69 @@ public static MeterProviderBuilder AddInMemoryExporter(this MeterProviderBuilder
             return AddInMemoryExporter(builder, exportedItems, new MetricReaderOptions(), configureMetricReader);
         }
 
+        /// <summary>
+        /// Adds InMemory metric exporter to the <see cref="MeterProviderBuilder"/> using default options.
+        /// </summary>
+        /// <param name="builder"><see cref="MeterProviderBuilder"/> builder to use.</param>
+        /// <param name="exportFunc">A function which will replace <see cref="InMemoryExporter{T}.Export(in Batch{T})"/>.</param>
+        /// <param name="configureMetricReader"><see cref="MetricReader"/> configuration options.</param>
+        /// <returns>The instance of <see cref="MeterProviderBuilder"/> to chain the calls.</returns>
+        public static MeterProviderBuilder AddInMemoryExporter(
+            this MeterProviderBuilder builder,
+            Func<Batch<Metric>, ExportResult> exportFunc,
```

Comment:
Why Func rather than Action?

Reply:
I went back and forth on this detail a few times.

1. I think it's a reasonable ask, if a user is replacing the functionality of Export, making them responsible for the ExportResult as well.

2. Being able to control the return type offers more flexibility to unit test authors.
  I found one test that was dependent on `ExportResult.Failure`. 
    - **Before**: This test was taking advantage of a side effect of the `InMemoryExporter`, setting the collection to `null` causes the Export to return `Failure`.
https://github.com/open-telemetry/opentelemetry-dotnet/blob/a64128f896168e18b1d66fb8229130fd9ff8dbeb/test/OpenTelemetry.Tests/Trace/BatchExportActivityProcessorTest.cs#L190-L193
https://github.com/open-telemetry/opentelemetry-dotnet/blob/a64128f896168e18b1d66fb8229130fd9ff8dbeb/src/OpenTelemetry.Exporter.InMemory/InMemoryExporter.cs#L31-L36
    - **After**: Here the test can clearly define the behavior that it's validating. 
https://github.com/open-telemetry/opentelemetry-dotnet/blob/26d7e1fc30332a0aa17c32a287e156d0091bc050/test/OpenTelemetry.Tests/Trace/BatchExportActivityProcessorTest.cs#L193

Reply:
Makes sense to me, thanks!